openshift/mlflow-experiment-job-template.yaml

```diff
@@ -0,0 +1,52 @@
+apiVersion: v1
+kind: Template
+
+labels:
+  application: ${APPLICATION_NAME}
+
+metadata:
+  name: mlflow-experiment-job-template
+
+parameters:
+- description: Application name
+  value: 'mlflow-experiment'
+  name: APPLICATION_NAME
+  required: true
+- name: MLFLOW_TRACKING_URI
+  value: http://mlflow-server-test-route-aiops-kubeflow.cloud.paas.upshift.redhat.com/
```

Comment:
if we're running the experiment in the same namespace as the mlflow server, I think we can use internal routing. 
`mlflow-server-test-service.aiops-kubeflow.svc`
Maybe it even works without the `aiops-kubeflow.svc` suffix.

Or look at these environment variables, which are defined for every pod
```
LFLOW_SERVER_SVC_PORT=tcp://172.56.132.180:5000
MLFLOW_SERVER_SVC_PORT_5000_TCP=tcp://172.56.132.180:5000
MLFLOW_SERVER_SVC_PORT_5000_TCP_ADDR=172.56.132.180
MLFLOW_SERVER_SVC_PORT_5000_TCP_PORT=5000
MLFLOW_SERVER_SVC_PORT_5000_TCP_PROTO=tcp
MLFLOW_SERVER_SVC_SERVICE_HOST=172.56.132.180
MLFLOW_SERVER_SVC_SERVICE_PORT=5000
MLFLOW_SERVER_SVC_SERVICE_PORT_HTTP=5000
MLFLOW_SERVER_TEST_SERVICE_PORT=tcp://172.56.45.193:5000
MLFLOW_SERVER_TEST_SERVICE_PORT_5000_TCP=tcp://172.56.45.193:5000
MLFLOW_SERVER_TEST_SERVICE_PORT_5000_TCP_ADDR=172.56.45.193
MLFLOW_SERVER_TEST_SERVICE_PORT_5000_TCP_PORT=5000
MLFLOW_SERVER_TEST_SERVICE_PORT_5000_TCP_PROTO=tcp
MLFLOW_SERVER_TEST_SERVICE_SERVICE_HOST=172.56.45.193
MLFLOW_SERVER_TEST_SERVICE_SERVICE_PORT=5000
MLFLOW_SERVER_TEST_SERVICE_SERVICE_PORT_HTTP=5000
```

maybe we default to one of those, if `MLFLOW_TRACKING_URI` is not set, i.e. leave it blank here

Reply:
My recommendation is to make it required for the template to have the MLFLOW_TRACKING_URI set.

Reply:
but if we're running in the same namespace as the tracking server and we could infer the url of the tracking server that would be one parameter less. 
if we can't make it easily, then so be it...

Reply:
Should be fine. But I'm saying mlflow will look for that environment variable to be set so either way we will need to set that env var to have it communicate with the tracking server.

Reply:
I will change it  to `MLFLOW_TRACKING_URI=http://$(MLFLOW_SERVER_TEST_SERVICE_SERVICE_HOST):MLFLOW_SERVER_TEST_SERVICE_SERVICE_PORT` so it uses the internal route